### PR TITLE
Update to Gradle 7

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Update to Gradle 7
 
### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-java/issues/67
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dblock if you don't mind this one please, thank you

**Review by mitalawachat — DISMISSED** [review]

**Review by imRishN — DISMISSED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Review by vamshin — DISMISSED** [review]
LGTM! Thanks

**Comment by reta** [comment]
@vamshin @dblock sorry guys, mind taking a look? thank you

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,33 +1,5 @@
...
-# software distributed under the License is distributed on an
-# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-# KIND, either express or implied.  See the License for the
-# specific language governing permissions and limitations
-# under the License.
-#
-# Modifications Copyright OpenSearch Contributors. See
-# GitHub history for details.
-#
-
```
Comment (dblock) [inline gradle/wrapper/gradle-wrapper.properties:28]: Are we doing the right thing removing the ES license here? We typically leave these alone.

Comment (reta) [reply gradle/wrapper/gradle-wrapper.properties:28]: Hm ... unexpected glitch in the editor, sorry about that

Comment (reta) [reply gradle/wrapper/gradle-wrapper.properties:28]: Fixed, thanks @dblock !

**Review by dblock — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Comment by reta** [comment]
@vamshin @dblock could we merge this one? thank you!

**Pull request merged**